homeassistant/components/homekit_controller/button.py

```diff
@@ -46,17 +46,67 @@ class HomeKitButtonEntityDescription(ButtonEntityDescription):
 }
 
 
+class HomeKitEcobeeClearHoldButton(CharacteristicEntity, ButtonEntity):
+    """Representation of a Button control for Ecobee clear hold request."""
+
+    def __init__(
+        self,
+        conn,
+        info,
+        char,
+    ):
+        """Initialise a HomeKit button control."""
+        self.entity_description = ButtonEntityDescription(
+            key=CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD,
+            name="Clear Hold",
+            device_class=ButtonDeviceClass.UPDATE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+        )
+        super().__init__(conn, info, char)
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity is tracking."""
+        return [self.entity_description.key]
```

Comment:
`get_characteristic_types` drives polling and even subscription, but this type is write only. Can you test returning an empty list here. This is probably the first time we've tried that so i'm not 100% it will work. It might be you need to include it in this list to be able to use it with put_char, but i've been trying to avoid that in recent refactorings IIRC). If it doesn't work, ignore this request and i'll come back to it in the future. Obv better to try it while there is someone with a device to test it!

```suggestion
        return []
```

Reply:
This change only makes sense for write only characteristics - for preset and fan mode we want the number to be readable as well as settable. So only do this for ClearHold.